### PR TITLE
Modify Geant4 recipe to make possible to define an external directory for data sets, plus clean-up: 

### PR DESCRIPTION
**Pull request by ihrivnac**
- This PR restores back ca2863573252a2cefd93b52843e13c11932a1774
- Fix setting Geant4 data sets in build environment (failing in the previous version):
   - Removed setting data sets in env: section in geant4.sh
   - Added setting data sets environment via geant4-config in recipes with Geant4 tests
     (aliroot-guntest.sh, o2.sh)

### CODE REVIEW

**Comment by ihrivnac** [comment]
@ktf, @sawenzel , can you approve ? Thank you.

File: geant4.sh
Comment (sawenzel) [inline geant4.sh:113]: just to understand: Which module file is now setting these variables (or who is setting these things when we load Geant4 on the GRID?)

Comment (sawenzel) [reply geant4.sh:113]: Ah I see. You modify the modulefile in a post-step.

**Review by sawenzel — APPROVED** [review]

File: geant4.sh
Comment (ihrivnac) [reply geant4.sh:113]: The data paths are added in the next step.
Geant4 includes need not to be added in the ROOT_INCLUDE_PATH as we do not generate dictionaries from Geant4 classes.

**Comment by ihrivnac** [comment]
@ktf , can you merge ? Thank you.

**Comment by sawenzel** [comment]
@ktf: Shouldn't simulation experts be able to merge this as per CODEOWNERS? The button is not activated.

**Review by ktf — APPROVED** [review]

**Comment by ktf** [comment]
@sawenzel the issue was that since o2.sh and aliroot-gun.sh were modified, it required a review from `alisw@infrastructure-admins`. I added `alisw/pdp-experts` for all the "mostly PDP" things, so next time your review should be enough. Of course I could have added you to alisw@infrastructure-admin (or you can add yourself, since you should be a global admin), however I think it's better if we keep infrastructure-admin separate, so that we have some better chance to notice new recipes / recipes without a proper owner.

**Pull request merged**
